src/System.Net.HttpListener/tests/HttpListenerContextTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
+{
+    public class HttpListenerContextTests : IDisposable
+    {
+        private HttpListenerFactory Factory { get; }
+        private ClientWebSocket Socket { get; }
```

Comment:
I was planning on adding a bunch of WebSocket tests for functionality after this PR. For now it is used to create a context and is also disposed below.

There are also some tests in WebSocketTests.cs that perform a conversation - they're pretty minimal so I was hoping to expand on them